src/renderer/graph/utils.ts

```diff
@@ -155,69 +167,65 @@ const gainAtFrequency = (f: number, c: ITransferFuncCoeffs) => {
   return 10 * Math.log10(numerator / denominator);
 };
 
-export const getFilterPoints = (filter: IFilter) => {
-  const tf = getTFCoefficients(filter);
+// Get curve info for the preAmplification
+export const getPreAmpLine = (preAmp: number): IChartPointData[] =>
+  SAMPLE_FREQUENCIES.map((f) => {
+    return { x: f, y: preAmp };
+  });
 
-  const logStart = Math.log10(GRAPH_START);
-  const logEnd = Math.log10(GRAPH_END);
-  const step = (logEnd - logStart) / NUM_STEPS;
+// Get curve and point info for individual filters
+export const getFilterLineData = (filter: IFilter): IChartPointData[] => {
+  const tf = getTFCoefficients(filter);
 
-  // TODO: consider hard coding frequencies maybe
-  const frequencies = range(logStart, logEnd + step, step).map((p) => 10 ** p);
-  const data: ChartDataPoint[] = frequencies.map((f) => {
+  const data: IChartPointData[] = SAMPLE_FREQUENCIES.map((f) => {
     return { x: f, y: 0 };
   });
 
-  for (let i = 0; i < frequencies.length; i += 1) {
-    const freqFilterGain = gainAtFrequency(frequencies[i], tf);
+  for (let i = 0; i < SAMPLE_FREQUENCIES.length; i += 1) {
+    const freqFilterGain = gainAtFrequency(SAMPLE_FREQUENCIES[i], tf);
     data[i].y = freqFilterGain;
   }
 
   return data;
 };
 
-export const getTotalPoints = (
+// Get total curve info from filter and point data
+export const getCombinedLineData = (
   preAmp: number,
-  filterLines: ChartDataPoint[][]
+  filterLines: IChartLineDataPointsById
 ) => {
-  const logStart = Math.log10(GRAPH_START);
-  const logEnd = Math.log10(GRAPH_END);
-  const step = (logEnd - logStart) / NUM_STEPS;
-
-  const frequencies = range(logStart, logEnd + step, step).map((p) => 10 ** p);
-  const data: ChartDataPoint[] = frequencies.map((f) => {
+  const data: IChartPointData[] = SAMPLE_FREQUENCIES.map((f) => {
     return { x: f, y: 0 };
   });
 
-  for (let i = 0; i < frequencies.length; i += 1) {
+  for (let i = 0; i < SAMPLE_FREQUENCIES.length; i += 1) {
+    // Add fixed preamp gain for each sample point
     data[i].y = preAmp;
-    for (let j = 0; j < filterLines.length; j += 1) {
-      data[i].y += filterLines[j][i].y;
-    }
+    Object.values(filterLines).forEach((points) => {
+      // Add frequency gain obtained from each filter
+      data[i].y += points[i].y;
+    });
   }
 
   return data;
 };
 
+// Get total curve data points from state directly
 export const getDataPoints = (preAmp: number, filters: IFilter[]) => {
```

Comment:
I think we can rewrite this to use getFilterLineData()

Reply:
or get rid of it cuz it's not being used